rfc/system/1855-type-metadata.md

```diff
@@ -0,0 +1,128 @@
+# [RR] Typing Metadata
+
+**Authors:**
+
+- @maximsmol
+- @kennyworkman
+
+## 1 Executive Summary
+
+A small patch to introduce flytekit support for python-native type metadata.
+
+## 2 Motivation
+
+Users of flyte wish to have access to arbitrary parameter metadata specified in
+common docstring formats (eg.  ReST, Google, Numpydoc-style, etc.) after a
+workflows has been registered.
+
+There already exists property on the `LiteralType` model for metadata.
+
+The `LiteralType` definition in flyteidl already has a `metadata` field in its
+message definition
+(https://github.com/flyteorg/flyteidl/blob/master/protos/flyteidl/core/types.proto#L89),
+suggesting that support for metadata annotations for flyte types was in the
+works.
+
+We wish to add python-native support for this field by allowing flytekit to
+recognize and parse the use of
+[`typing.Annotation`](https://docs.python.org/3/library/typing.html#typing.Annotated).
+
+This could be to change the presentation of a form input ingesting typed values,
+surface descriptions to users or store indexes to control the order of
+parameters with custom frontends.
+
+An example of how we would use this annotation to tag a parameter with
+presentation information:
+
+
+```
+amplicon_min_alignment_score: Optional[
+        List[
+            Annotated[
+                int,
+                FlyteMetadata(
+                    {
+                        "appearance": {
+                            "multiselect": {
+                                "options": [50, 60, 70, 80, 90, 100],
+                                "allow_custom": True,
+                            }
+                        }
+                    }
+                ),
+            ]
+        ]
+    ],
+```
+
+## 3 Proposed Implementation
```

Comment:
Can we include the work here to strip them in [here](https://github.com/flyteorg/flytepropeller/blob/master/pkg/compiler/transformers/k8s/node.go#L213).

Reply:
can you help me understand why?

Reply:
this is so that the backend representation at runtime does not bloat and is important to remain efficient and performant